src/map/skill.c

```diff
@@ -17930,8 +17895,115 @@ static void skill_weaponrefine(struct map_session_data *sd, int idx)
 }
 
 /*==========================================
- *
+ * Auto Spell / Hindsight
  *------------------------------------------*/
+
+/**
+ * Returns the list of skills Auto Spell may use and their conditions.
+ * 
+ * The list is SORTED by the required autospell level (autospell_level field)
+ * 
+ * @param list pointer to receive the list's pointer (pointer to an array)
+ * @returns list length
+ */
+static int skill_autospell_list(const struct autospell_skill **list)
+{
+	nullpo_retr(0, list);
+
+	// MUST be sorted by autospell_level (first field)
+	static const struct autospell_skill skills_list[] = {
```

Comment:
I am really unsure whether this was the best approach for it. Feel free to give suggestions and discuss about it.

As I mentioned in the PR description, I did some sort of automated test to check if it still works, so if we have a better idea, we can validate it without too much effort.

Reply:
de-hardcode it... move to a file within `db/` as best approach

Reply:
I made a separate PR with the refactor + de-hardcode :) #3237 . Thanks!